### DISCUSSION TITLE
Motion safe by default

### DISCUSSION THREAD

**Discussion by dan-overton** [tailwindlabs/tailwindcss#12864] [Help]
Is there a way to use `motion-safe` by default when using animations? i.e. by adding it to the theme?
I'd prefer not to have to remember to add it every time I add an animation to an element.

Apologies if this is in the docs, I couldn't see any obvious answer from looking around.

**Comment by wongjn** [comment]
You could [create a Tailwind plugin](https://tailwindcss.com/docs/plugins#adding-utilities) which is a copy of the the core `animation` Tailwind plugin:

https://github.com/tailwindlabs/tailwindcss/blob/7361468f77500105b0559e879e121f34306e8da2/src/corePlugins.js#L993-L1023

And add the `motion-safe`-equivalent media query to the generated rules.

**Comment by emattias** [comment]
I think wrapping transition and animation related classes in the `motion-safe` media query should be the default (with a variant to not have the mq) or atleast an option to make it so.

I implemented this plugin (using tw v3). 

Since I couldnt find a way to make tw not add the default non mq version I had to add non mq overrides that set them to `initial` (for when the user doesnt have reduced motion enabled):

![CleanShot 2025-05-21 at 11 23 14@2x](https://github.com/user-attachments/assets/6327ddb9-cdd6-47ae-b0fd-01eead4c3654)

 If someone knows of a way to stop tw from adding the default non mq wrapped transition and animation classes, let me know.

```js
plugins: [
    plugin(function ({ addUtilities, theme }) {
      const config = theme('transitionDuration');
      const transitions = theme('transitionProperty');
      const animations = theme('animation');
      const timingFunctions = theme('transitionTimingFunction');

      const newUtilities = {};
      const initialUtilities = {}; // To store the non-media query versions
      const resetVal = 'initial';

      Object.keys(transitions).forEach((key) => {
        const cssKey =
          key === 'DEFAULT' ? 'transitionProperty' : `transition-${key}`;
        newUtilities[`.${cssKey}`] = {
          '@media (prefers-reduced-motion: no-preference)': {
            'transition-property': transitions[key],
            'transition-timing-function': timingFunctions['DEFAULT'],
            'transition-duration': config['DEFAULT'],
          },
        };
        initialUtilities[`.${cssKey}`] = {
          'transition-property': resetVal,
          'transition-timing-function': resetVal,
          'transition-duration': resetVal,
        };
      });

      Object.keys(animations).forEach((key) => {
        newUtilities[`.animate-${key}`] = {
          '@media (prefers-reduced-motion: no-preference)': {
            animation: animations[key],
          },
        };
        initialUtilities[`.animate-${key}`] = {
          animation: resetVal,
        };
      });

      Object.keys(config).forEach((key) => {
        const cssKey =
          key === 'DEFAULT' ? 'transitionDuration' : `duration-${key}`;
        newUtilities[`.${cssKey}`] = {
          '@media (prefers-reduced-motion: no-preference)': {
            'transition-duration': config[key],
          },
        };
        initialUtilities[`.${cssKey}`] = {
          'transition-duration': resetVal,
        };
      });

      const defaultTimingFunction = timingFunctions['DEFAULT'];
      newUtilities['.ease-default'] = {
        '@media (prefers-reduced-motion: no-preference)': {
          'transition-timing-function': defaultTimingFunction,
        },
      };
      initialUtilities['.ease-default'] = {
        'transition-timing-function': resetVal,
      };
      addUtilities(initialUtilities); // Add the non-media query versions with reset values

      addUtilities(newUtilities, {
        variants: ['responsive', 'hover', 'focus'],
      });
    })
]
```

**Comment by MangoMarcus** [comment]
I thought about this but there are animations that I don't believe apply to motion-safe like opacity, right? Ie. animations without motion. So you wouldn't want transition-opacity to be motion-safe by default
